src/containerapp/azext_containerapp/daprcomponent_resiliency_decorator.py

```diff
@@ -0,0 +1,375 @@
+                safe_set(self.component_resiliency_patch_def, "properties", "outboundPolicy", "httpRetryPolicy", "retryBackOff", "maxIntervalInMilliseconds", value = DEFAULT_COMPONENT_HTTP_RETRY_BACKOFF_MAX_DELAY)
+
+        safe_set(self.component_resiliency_patch_def, "properties", "outboundPolicy", "timeoutPolicy", "responseTimeoutInSeconds", value = (self.get_argument_out_timeout_response() or safe_get(component_resiliency_def, "properties", "outboundPolicy", "timeoutPolicy", "responseTimeoutInSeconds")))
+
+        if safe_get(self.component_resiliency_patch_def, "properties", "inboundPolicy", "httpRetryPolicy", "maxRetries") or safe_get(self.component_resiliency_patch_def, "properties", "inboundPolicy", "httpRetryPolicy", "retryBackOff", "initialDelayInMilliseconds") or safe_get(self.component_resiliency_patch_def, "properties", "inboundPolicy", "httpRetryPolicy", "retryBackOff", "maxIntervalInMilliseconds"):
+            safe_set(component_resiliency_def, "properties", "inboundPolicy", "httpRetryPolicy", value = safe_get(self.component_resiliency_patch_def, "properties", "inboundPolicy", "httpRetryPolicy"))
+        if safe_get(self.component_resiliency_patch_def, "properties", "inboundPolicy", "timeoutPolicy", "responseTimeoutInSeconds"):
+            safe_set(component_resiliency_def, "properties", "inboundPolicy", "timeoutPolicy", value = safe_get(self.component_resiliency_patch_def, "properties", "inboundPolicy", "timeoutPolicy"))
+        if safe_get(self.component_resiliency_patch_def, "properties", "outboundPolicy", "httpRetryPolicy", "maxRetries") or safe_get(self.component_resiliency_patch_def, "properties", "outboundPolicy", "httpRetryPolicy", "retryBackOff", "initialDelayInMilliseconds") or safe_get(self.component_resiliency_patch_def, "properties", "outboundPolicy", "httpRetryPolicy", "retryBackOff", "maxIntervalInMilliseconds"):
+            safe_set(component_resiliency_def, "properties", "outboundPolicy", "httpRetryPolicy", value = safe_get(self.component_resiliency_patch_def, "properties", "outboundPolicy", "httpRetryPolicy"))
+        if safe_get(self.component_resiliency_patch_def, "properties", "outboundPolicy", "timeoutPolicy", "responseTimeoutInSeconds"):
+            safe_set(component_resiliency_def, "properties", "outboundPolicy", "timeoutPolicy", value = safe_get(self.component_resiliency_patch_def, "properties", "outboundPolicy", "timeoutPolicy"))
+
+        component_resiliency_def = clean_null_values(component_resiliency_def)
+        self.component_resiliency_update_def = component_resiliency_def
+    
+    def update(self):
+        try:
+            r = self.client.create_or_update(
+            cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+            name=self.get_argument_name(), dapr_component_name=self.get_argument_dapr_component_name(),
+            environment_name=self.get_argument_environment(), component_resiliency_envelope=self.component_resiliency_update_def,
```

Comment:
Should this be using the patch def?

Reply:
We don't have patch for dapr component :(